### PR TITLE
T/473: Remove gulp dependency

### LINKED ISSUE #473: Consider switching from gulp to npm scripts to reduce bloat
We have `ckeditor5` repository and `ckeditor5-*` sub repositories. Right now all of them use gulp.

As for `ckeditor5` itself, it doesn't matter that much, but we already have some npm scripts there, so it's a bit messy. We could try to standardise this.

A bigger issue is with all the other packages. All of them base on gulp which isn't the most lightweight dependency and we use it to expose two simple tasks – `lint` and `lint-staged`. 

So, every `ckeditor5-*` package depends on `@ckeditor/ckeditor5-dev-lint`, `gulp`, `guppy-pre-commit` which have around 380 dependencies: https://gist.github.com/Reinmar/d9633ea5bf583c6a85111a269d273995.

We install them using Lerna and only `guppy-pre-commit` is installed in every package physically. But it still takes a time to calculate all the deps for every package to see what needs to be hoisted to `ckeditor5`.

Another thing is that we have one additional file – `gulpfile.js` that we could remove if we switched to npm scripts. And, we could drop the `ckeditor5-dev-lint` package altogether if we'd be lucky.

So, the plan would be to try to implement:

* three scripts `lint`, `lint-staged` and `pre-commit`,
* automatic git pre-commit hook installation (just like `guppy-pre-commit` but triggering npm instead of gulp).

It seems that it can be easily done using https://github.com/okonet/lint-staged. We just need to figure out whether this will be any real improvement in terms of dependency bloat.

### PR DESCRIPTION
**Pull request by jodator**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Remove gulp dependency in favor of npm scripts. Closes #473.

BREAKING CHANGE: Gulp task are removed. New npm scripts were introduced for dev related tasks.

---

### Additional information

* Requires https://github.com/ckeditor/ckeditor5-dev/pull/298.
* Most of gulp tasks were refactored to js scripts (located in `./scripts/release` and `./scripts/docs`) and are invoked by appropriate npm `run-script`.

### CODE REVIEW

File: scripts/docs/build-docs.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* eslint-env node */
+
+'use strict';
+
+const assertIsInstalled = require( './../util/assertisinstalled' );
```
Comment (Reinmar) [inline scripts/docs/build-docs.js]: Let's call this directory `utils/` to be consistent.

**Comment by pomek** [comment]
I've just checked all the tasks. All works fine.

**Comment by Reinmar** [comment]
@szymonkups I don't want to merge this PR and leave, but it LGTM too. I think you can merge it but please check the docs for some leftovers, cause I haven't checked that bit.

**Comment by szymonkups** [comment]
I've checked both PR's and everything seems fine. But I will not close them: after merging PR in `ckeditor5-dev` we need to release packages, update their versions in `package.json` in this PR. After that we will be able to safely merge this PR.

**Pull request merged**